metagraph/integration_tests/test_transform_anno.py

```diff
@@ -104,9 +108,11 @@ def _check_aggregation_min_max_value(self, min_count, max_value, expected_densit
         self.assertEqual(res.returncode, 0)
         out = res.stdout.decode().split('\n')[2:]
         self.assertEqual('labels:  1', out[0])
-        self.assertEqual('objects: 46960', out[1])
-        self.assertEqual(f'density: {expected_density}', out[2])
-        self.assertEqual(f'representation: {self.anno_repr}', out[3])
+        split = out[2].split()
+        self.assertEqual(out[2], split[0] + ' ' + split[1])
+        density = float(split[1])
+        self.assertIn(out[1], ['objects: 47633', 'objects: 46960'])
+        self.assertLess(abs(density - expected_density * 46960 / 47633), 10**-6)
```

Comment:
As far as I can tell, all failing integration tests were seemingly failing due to `annotations.num_objects()` now returning the number of BOSS edges, including dummy ones, and `print_annotation_stats` doesn't seem to contain any information about the initial graph to use `graph.num_nodes()` for the computations instead. Is it okay to update integration tests like this, or should I propagate `num_nodes()` into annotations somehow when it's different from `num_objects()`?

Reply:
You can include both a graph and an annotation into a `metagraph stat` run and it'll print both. So I think the best thing to do is to update the `stat` print-out for a graph to also print `max_index()`, then you can check if that matches the annotation's `num_objects()`

Reply:
Okay, I refactored _get_stats to return a dict, and also added checks based on max_index and its scale related to num_nodes into integration tests. Hope they'll pass now :smiling_face_with_tear: